src/ProtocolStreamReader.php

```diff
@@ -30,33 +30,32 @@ public function __construct($input)
     {
         $this->input = $input;
         Loop\addReadStream($this->input, function() {
-            while (($c = fgetc($this->input)) !== false) {
-                $this->buffer .= $c;
-                switch ($this->parsingMode) {
-                    case ParsingMode::HEADERS:
-                        if ($this->buffer === "\r\n") {
-                            $this->parsingMode = ParsingMode::BODY;
-                            $this->contentLength = (int)$this->headers['Content-Length'];
-                            $this->buffer = '';
-                        } else if (substr($this->buffer, -2) === "\r\n") {
-                            $parts = explode(':', $this->buffer);
-                            $this->headers[$parts[0]] = trim($parts[1]);
-                            $this->buffer = '';
+            $c = fgetc($this->input);
```

Comment:
This is the tricky part... There are multiple problems with the way the stream is currently read. And one of those is going to be very hard to fix...

You are looping while `fgetc() !== false`, which tests only for EOF (stream closed). So the loop would go on forever, because it simply waits if no new data is available.

Normally we _could_ set the STDIN stream to nonblocking with `stream_set_blocking(STDIN, false)` Then `fgetc` would simply return an empty string if no new data is available instead of blocking.

But! There is a bug/weird behaviour in windows that makes it impossible to set STDIN to non-blocking.

We could maybe call `stream_select` with a zero timeout before **any** call to `fgetc` but I don't know the performance implications of this.

Otherwise, we would have to spawn a second process/thread to only read from STDIN and pass the data via a socket, which can be non-blocking on windows too.